### PR TITLE
fix: menu buttons resseting scroll position on mobile

### LINKED ISSUE #6667: Menu buttons resseting scroll position on mobile
### What happened?

Also happens when clicking profile and changing chain: 

https://github.com/kodadot/nft-gallery/assets/36627808/38cf92a1-7cd8-48f1-b7fe-628b19d58360

### Please reproduce in steps

Mobile -> scroll down -> open menu -> cart

### PR DESCRIPTION
**Pull request by stephenjason89**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #6667 #6669 #6668 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=16UcV9V6nVvPYdHz98ymUKmNLkzjCEU5sbKJMi7hxYyTHjzR&usdamount=100&donation=true)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 769abca</samp>

Refactored the code for opening the shopping cart modal from the navbar. Moved the logic and configuration from `ShoppingCartModalConfig.ts` and `ShoppingCartButton.vue` to `Navbar.vue` to improve code organization and performance.

### <samp>🤖 Generated by Copilot at 769abca</samp>

> _`ShoppingCartButton`_
> _Simpler, no more `emit` -_
> _Autumn leaves falling_

### CODE REVIEW

File: components/Navbar.vue [hunk 1]
```diff
@@ -140,7 +140,6 @@
             data-cy="chain-select" />
 
           <NotificationBoxButton
-            v-if="account"
```
Comment (roiLeo) [inline components/Navbar.vue:143]: why remove this?

File: components/Navbar.vue [hunk 2]
```diff
@@ -255,7 +254,6 @@ const openWalletConnectModal = (): void => {
 }
 
 const showMobileNavbar = () => {
-  document.body.classList.toggle('is-clipped')
```
Comment (roiLeo) [inline components/Navbar.vue:258]: There is a reason why I added this line but I can't remember why (ref #6382)

Comment (roiLeo) [reply components/Navbar.vue:258]: Ok now I remember, on Mobile if you remove this line, you can still scroll through explore page even if sidebar is open & card menu header is looking bad

Screenshot 2023-08-11 at 10-37-27 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer

**Review by roiLeo — CHANGES_REQUESTED** [review]
https://github.com/kodadot/nft-gallery/pull/6673/files#r1291063287

File: components/Navbar.vue (hunk 1 above)
Comment (stephenjason89) [reply components/Navbar.vue:143]: Sorry, i don't have an account so i removed it to test. But mistakenly committed it. Was about to change it when i saw your comment. You are too fast @roiLeo haha

File: components/Navbar.vue (hunk 2 above)
Comment (stephenjason89) [reply components/Navbar.vue:258]: oh ok @roiLeo , will check it out further

Comment (stephenjason89) [reply components/Navbar.vue:258]: @roiLeo I can confirm that without the is-clipped toggle, you can still scroll through explore page even if the sidebar menu is opened. 

If we want the is-clipped toggle then, I can save the scroll location before the sidebar opens and restore it when sidebar closes. Is this okay?

Comment (stephenjason89) [reply components/Navbar.vue:258]: the lastScrollPosition is already saved here so i just restored the position everytime the sidebar closes.

Comment (roiLeo) [reply components/Navbar.vue:258]: Still same issue for me

Comment (stephenjason89) [reply components/Navbar.vue:258]: Here's a recording of mine with the latest deployment on netlify

https://deploy-preview-6673--koda-canary.netlify.app/

https://github.com/kodadot/nft-gallery/assets/3810177/c2796b6d-fadc-4b1c-9212-265ae2d37e02

Might be a browser issue?
What browser are you using?

Comment (stephenjason89) [reply components/Navbar.vue:258]: image

Comment (stephenjason89) [reply components/Navbar.vue:258]: code is very straight forward
document.documentElement.scrollTop = lastScrollPosition.value

It should be supported by almost every browsers

Comment (roiLeo) [reply components/Navbar.vue:258]: cleared cache now it work

**Review by roiLeo — APPROVED** [review]
code lgtm

**Comment by prury** [comment]
Notifications ✔️ 
Shopping Cart ✔️ 
Profile ✔️ 

For chain switching lets wait for #6613

**Comment by yangwao** [comment]
pay 60 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $60 @ 5.03 USD/DOT ~ 11.928 $DOT
🧗 [16UcV9V6nVvPYdHz98ymUKmNLkzjCEU5sbKJMi7hxYyTHjzR](https://polkadot.subscan.io/account/16UcV9V6nVvPYdHz98ymUKmNLkzjCEU5sbKJMi7hxYyTHjzR)
🔗 [0x7a70f99824f21f6d60bdb3c7a4e993ed935c0bd6b41f9b7c844f556e69fabc25](https://polkadot.subscan.io/extrinsic/0x7a70f99824f21f6d60bdb3c7a4e993ed935c0bd6b41f9b7c844f556e69fabc25)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**